Parse/src/main/java/com/parse/ParseDecoder.java

```diff
@@ -122,7 +122,7 @@ public Object decode(Object object) {
     }
 
     if (typeString.equals("Object")) {
-      return ParseObject.fromJSON(jsonObject, null, true, this);
+      return ParseObject.fromJSON(jsonObject, null, !jsonObject.has(ParseObject.KEY_SELECTED_KEYS), this);
```

Comment:
it seems to me that fromJSON() should figure out whether things are complete instead of here right?

Just trying to get KEY_SELECTED_KEYS out of being a public static variable..